### ISSUE TITLE
Rails 4.1 Support

### ISSUE THREAD

**Issue by pat** [pat/joiner#1]
TL;DR: Get [this spec](https://github.com/pat/joiner/blob/master/spec/acceptance/joiner_spec.rb) green on Rails 4.1.

This gem contains the logic around using SQL joins via ActiveRecord classes. In Rails 4.1.0 (at least, in the current beta), the logic around this has changed dramatically, and getting it working is going to require a completely new approach in `Joiner::Joins` (here is [the existing approach](https://github.com/pat/joiner/blob/master/lib/joiner/joins.rb)).

I pulled the logic from Thinking Sphinx out into this gem so Thinking Sphinx can continue on its merry way, and once this gem supports Rails 4.1.0, I'll release a new version with the appropriate dependency and it'll just slot into place neatly.

The requirement of any changes is that `Joiner::Joins` continues to have the following methods (and corresponding behaviours):

``` ruby
def initialize(model)
   # takes the model that the SQL query is based upon
end

def add_join_to(path)
  # Adds an OUTER join for the given association path
end

def alias_for(path)
  # Returns the table alias for the given association path
end

def join_values
  # Returns the underlying join values - whatever's returned by this method must be
  # accepted as an argument to ActiveRecord::Base.joins.
end
```

For clarity on what a path is, have a look at this project's [README](https://github.com/pat/joiner#usage).

The `rails41` branch is currently set up to test against ActiveRecord 4.1.0.beta1. If you want to give it a spin and see if you can get anywhere with a working implementation, be my guest. There are both unit and acceptance specs - and the acceptance specs are (correctly) breaking at the moment:

``` term
git clone git://github.com/pat/joiner.git
git checkout origin/rails41 -b rails41
bundle
bundle exec rspec
```

Any pull requests _greatly_ appreciated.
